distributed/scheduler.py

```diff
@@ -2712,115 +2714,122 @@ def clear_task_state(self):
         average expected load.
         """
         with log_errors():
-            if keys:
-                tasks = {self.tasks[k] for k in keys}
-                missing_data = [ts.key for ts in tasks if not ts.who_has]
-                if missing_data:
-                    return {"status": "missing-data", "keys": missing_data}
-            else:
-                tasks = set(self.tasks.values())
-
-            if workers:
-                workers = {self.workers[w] for w in workers}
-                workers_by_task = {ts: ts.who_has & workers for ts in tasks}
-            else:
-                workers = set(self.workers.values())
-                workers_by_task = {ts: ts.who_has for ts in tasks}
+            async with self._lock:
```

Comment:
I am new to async so please excuse my ignorance: Documentation (https://docs.python.org/3.6/library/asyncio-sync.html) proposes to use this snippet instead:

```
lock = Lock()
...
with (yield from lock):
    ...
```

Is there a functional difference that made you prefer one over the other?

Reply:
`async with` is a newer syntax that wasn't available until 3.5 I think.  It is usable within async def functions while `yield from` syntax is usable from within `@asyncio.coroutines`.  In general the Dask codebase switched from coroutiens to async def functions a while ago, and so we tend to prefer to use that syntax.